src/applications/disability-benefits/526EZ/components/AsyncDisplayWidget.jsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+
+import LoadingIndicator from '@department-of-veterans-affairs/formation/LoadingIndicator';
+import AlertBox from '@department-of-veterans-affairs/formation/AlertBox';
+
+const PENDING = 'PENDING';
+const RESOLVED = 'RESOLVED';
+const REJECTED = 'REJECTED';
+
+
+/**
+ * Handles the various display statuses when calling an asynchronous function.
+ */
+export default class AsyncDisplayWidget extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: null,
+      promiseState: PENDING
+    };
+
+    // Make sure we configured it correctly
+    // StringField passes 'ui:options' as the options prop
+    const uiOptions = props.options;
+    if (!uiOptions) {
+      throw new Error('No ui:options supplied to AsyncDisplayWidget.');
+    }
+
+    if (typeof uiOptions.viewComponent !== 'function') {
+      throw new Error('AsyncDisplayWidget requires viewComponent in ui:options to be a React element.');
+    }
+
+    if (typeof uiOptions.callback !== 'function') {
+      throw new Error('AsyncDisplayWidget requires callback in ui:options to be a function.');
+    }
+
+    // TODO: Don't call the callback _every_ time the component is mounted
+    const cbPromise = uiOptions.callback();
+    if (cbPromise instanceof Promise) {
+      cbPromise.then((data) => {
+        this.setState({
+          data,
+          promiseState: RESOLVED
+        });
+      }).catch((data) => {
+        if (data instanceof Error) {
+          throw data;
+        }
+
+        this.setState({
+          data,
+          promiseState: REJECTED
+        });
+      });
+    } else {
+      throw new Error(`AsyncDisplayWidget: Expected callback to return a Promise, but got ${typeof cbPromise}.`);
+    }
+  }
+
+
+  // Not sure if this will be useful yet
```

Comment:
You'll need to add something to make sure you don't call `setState` if the promise resolves after the component is unmounted. In the past, I've set an instance variable called `unmounted` in `componentDidUnmount` and checked to make sure that wasn't set before calling `setState` when the promise resolves.